elements/pl-order-blocks/pl-order-blocks.py

```diff
@@ -0,0 +1,414 @@
+
+    if pl.get_string_attrib(element, 'grading-method', 'ordered') != 'external' and len(correct_answers) == 0:
+        raise Exception('There are no correct answers specified for this question.')
+
+    if (correct_answers_ranking != sorted(correct_answers_ranking)):
+        # sort correct answers by indices specified in corect_answers_ranking
+        correct_answers = [x for _, x in sorted(zip(correct_answers_ranking, correct_answers))]
+
+    minIncorrect = pl.get_integer_attrib(element, 'min-incorrect', None)
+    maxIncorrect = pl.get_integer_attrib(element, 'max-incorrect', None)
+
+    if ((minIncorrect is None) & (maxIncorrect is None)):
+        mcq_options = correct_answers + incorrect_answers
+    else:
+        # Setting default for min-correct and checking for correct interval
+        if minIncorrect is None:
+            minIncorrect = 1
+        else:
+            if minIncorrect > len(incorrect_answers):
+                raise Exception('min-incorrect must be smaller than the number of given distractors.')
+        # Setting default for max-correct and checking for correct interval
+        if maxIncorrect is None:
+            maxIncorrect = len(incorrect_answers)
+        else:
+            if maxIncorrect > len(incorrect_answers):
+                raise Exception('max-incorrect must be smaller than the number of given distractors.')
+        if minIncorrect > maxIncorrect:
+            raise Exception('min-incorrect must be smaller than max-incorrect.')
+        incorrect_answers_count = random.randint(minIncorrect, maxIncorrect)
+        mcq_options = correct_answers + random.sample(incorrect_answers, incorrect_answers_count)
+
+    is_shuffle = pl.get_boolean_attrib(element, 'shuffle-source-blocks', SHUFFLE_MCQ_OPTIONS_DEFAULT)  # default to FALSE, no shuffling unless otherwise specified
+
+    if is_shuffle is True:
+        random.shuffle(mcq_options)
+    else:
+        mcq_options = html_ordering
+
+    data['params'][answer_name] = mcq_options
+    data['correct_answers'][answer_name] = {'correct_answers': correct_answers,
+                                            'correct_answers_indent': correct_answers_indent}
+
+
+def render(element_html, data):
+    element = lxml.html.fragment_fromstring(element_html)
+    answer_name = pl.get_string_attrib(element, 'answers-name')
+
+    if data['panel'] == 'question':
+        mcq_options = []   # stores MCQ options
+        student_previous_submission = []
+        submission_indent = []
+
+        for html_tags in element:
+            if html_tags.tag == 'pl-answer':
+                mcq_options.append(html_tags.text)   # store the original specified ordering of all the MCQ options
+
+        answer_name = pl.get_string_attrib(element, 'answers-name')
+        source_header = pl.get_string_attrib(element, 'source-header', 'Drag from here:')
+        solution_header = pl.get_string_attrib(element, 'solution-header', 'Construct your solution here:')
```

Comment:
Can you please check all cases of using `pl.get_*_attrib()` to make sure we are using `*_DEFAULT` variables defined at the top of this file for the default values, and not hard-coding them? I haven't checked the whole file, but it looks like there are currently a lot more attributes than there are DEFAULT variables 😄

Reply:
fixed